shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -262,11 +262,21 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
       _activeModel = std::make_unique<flutter::TextInputModel>();
     }
   } else if ([method isEqualToString:kShowMethod]) {
+    // Ensure the plugin is in hierarchy. Only do this with accessibility disabled.
+    // When accessibility is enabled cocoa will reparent the plugin inside
+    // FlutterTextField in [FlutterTextField startEditing].
+    if (_client == nil) {
+      [_flutterViewController.view addSubview:self];
+    }
+    [self.window makeFirstResponder:self];
     _shown = TRUE;
-    [_textInputContext activate];
   } else if ([method isEqualToString:kHideMethod]) {
+    // With accessiblity enabled TextInputPlugin is inside _client, so take the
+    // nextResponder from the _client.
+    NSResponder* nextResponder = _client != nil ? _client.nextResponder : self.nextResponder;
+    [self.window makeFirstResponder:nextResponder];
+    [self removeFromSuperview];
```

Comment:
This is probably also needed in `kClearClientMethod`. It's not guaranteed that we'll get a `kHideMethod` when the connection is closed by the framework.

Reply:
Done.